### ISSUE TITLE
Wagtail fails to validate unique_together in InlinePanel

### ISSUE THREAD

**Issue by m1kola** [wagtail/wagtail#4360]
### Issue Summary

Wagtail fails to validate `unique_together` on related models.

### Steps to Reproduce

1. Define a model like

    ```python
    class MyPage(Page):
        content_panels = Page.content_panels + [
            InlinePanel('options', label=_('Options')),
        ]

    OPTION_CHOICES = (
        ('one', 'Option 1'),
        ('two', 'Option 2'),
    )

    class RelatedProgram(Orderable):
        source_page = ParentalKey(MyPage, models.CASCADE, related_name='options')
        option = models.CharField(max_length=255, choices=OPTION_CHOICES)

        panels = [
            FieldPanel('program'),
        ]

        class Meta:
            unique_together = ('source_page', 'option')

    ```
2. In the admin UI try to create or edit a page and specify the same option

Result: 

```
IntegrityError: duplicate key value violates unique constraint "myapp_mypage_source_page_id_option_something_uniq
```

### Technical details

* Python version: 3.6
* Django version: 1.11.10.
* Wagtail version: 1.13.1

**Comment by gasman** [comment]
Hi @m1kola,
This was fixed in django-modelcluster 4.0: https://github.com/wagtail/django-modelcluster/pull/84

(I _think_ it should be possible to run django-modelcluster 4.0 against Wagtail 1.13.1, even though Wagtail specifies <4.0; - the major version bump was only due to dropping Django 1.8-1.10 support. Upgrading to Wagtail 2.0 is the preferred fix, of course :-) )

**Comment by m1kola** [comment]
Oh, good to know. I haven't had time to upgrade this project to Wagtail 2.0, but I'll definitely do that. Hope it will be soon :)

Thanks Matthew 🥇
